### ISSUE TITLE
lbsslcertkey: only send nodomaincheck on certificate change

### ISSUE THREAD

**Issue by ajsiersema** [citrix/terraform-provider-netscaler#39]
**Issue description**
lbsslcertkey won't accept the `nodomaincheck ` attribute/argument on creating a certificate.
`nodomaincheck`  is however needed in most cases to 'refresh' a certificate.

**Steps to reproduce**

Try this in a resources file.

```
resource "netscaler_sslcertkey" "lbsslcertkey" {
  certkey = "TLS_${lookup(var.vip, "name")}"
  cert = "/nsconfig/ssl/${lookup(var.vip, "name")}.cert"
  key = "/nsconfig/ssl/${lookup(var.vip, "name")}.key"
  expirymonitor = "DISABLED"
  nodomaincheck = true
}
```

Expected results : certificate created.
Actual results     : certificate creation fails.

Remove nodomaincheck, upload new certificate files and rerun : success.

**Further background information**
The NITRO REST API distinguishes between `add` and `change` actions:
https://developer-docs.citrix.com/projects/netscaler-nitro-api/en/12.0/configuration/ssl/sslcertkey/sslcertkey/#operations
where unfortunately ADD won't accept nodomaincheck, but change will.

Please detect the presence of a sslcertkey and send either add or change and only nodomaincheck on change.

terraform 12 supports conditional null arguments, I'm not sure if this would solve the issue, and in any case, we're not running v12 + supported netscaler provider :)

**Comment by giorgos-nikolopoulos** [comment]
I was able to reproduce the error.

As you correctly pointed out the root cause is the fact that NITRO API will accept `nodomaincheck` only on an update and not on creation of the `sslcertkey` resource.

I am working on a fix which will basically detect this condition, meaning having an update only argument on a creation operation, and will enforce a workflow where the resource is first created with all acceptable arguments and subsequently, without needing user intervention, it will do an update operation which will contain the attributes left out from the creation operation.

This should work fine with terraform v0.11.

I will post updates here when this fix is ready.

**Comment by giorgos-nikolopoulos** [comment]
A little elaboration on the fix and the resource behavior with the fix.

The `nodomaincheck` argument is not a resource attribute but rather a flag for the change operation of the resource.

As such the fix ignores it completely when creating a new sslcertkey instance.
When a resource is changed it is taken into account if its value is `true`.

In the local state shown by `terraform show` the user will always see a `nodomaincheck = false` value.
This is a bit misleading since the actual resource on Citrix ADC does not have this attribute.

This value is there so that when changing the resource a value of `nodomaincheck = true` is passed to the change request if it is so set in the resource description file.

So in short:
* `nodomaincheck` is only used when the resource operation is relevant.
* local state will always show a value of `nodomaincheck = false` in order to honor any `nodomaincheck = true` in the resource file e.g. `resources.tf`

@ajsiersema 

I have made a release with this fix. You can find it [here](https://github.com/citrix/terraform-provider-netscaler/releases).

Let us know if this works out for you.

**Comment by ajsiersema** [comment]
Works like a charm. Thank you very much for fixing this. Saves ugly templating. I now have a single resources file for load balancers with and without SSL.
